optimism/src/main.rs

```diff
@@ -62,14 +62,24 @@ pub fn main() -> ExitCode {
         ark_ec::short_weierstrass_jacobian::GroupAffine<ark_bn254::g1::Parameters>,
     >::default();
 
-    let new_chunk = || proof::WitnessColumns {
+    let reset_chunk = |witness_columns: &mut proof::WitnessColumns<Vec<_>>| {
+        let proof::WitnessColumns {
+            scratch,
+            instruction_counter,
+            error,
+        } = witness_columns;
+        // Resize without deallocating
+        scratch.iter_mut().for_each(Vec::clear);
```

Comment:
Oh, so then it is assumed that since each chunk will have the same size, it is fine if the same memory allocations get "nullified" instead of deallocated. And then they can be reassigned again to the new values? Nice